woodwork/table_accessor.py

```diff
@@ -84,70 +113,81 @@ def init(self,
-                warnings.warn("A schema was provided and the following parameters were ignored: " + ", ".join(extra_params), ParametersIgnoredWarning)
-
-        else:
-            # Perform type inference and update underlying data
-            parsed_logical_types = {}
-            for name in self._dataframe.columns:
-                series = self._dataframe[name]
-
-                logical_type = None
-                if logical_types:
-                    logical_type = logical_types.get(name)
-
-                logical_type = _get_column_logical_type(series, logical_type, name)
-                parsed_logical_types[name] = logical_type
-
-                updated_series = logical_type.transform(series)
-                if updated_series is not series:
-                    self._dataframe[name] = updated_series
-
-            column_names = list(self._dataframe.columns)
-
-            # TableSchema uses a different default for use_standard_tags so we need to define it here
+        def left_join(left: dict, right: dict):
+            left = left or {}
+            right = right or {}
+            combined_dict = {}
+            for key in right:
+                combined_dict[key] = right[key]
+            for key in left:
+                combined_dict[key] = left[key]
+            return combined_dict
+
+        def normalize_standard_tags(use_standard_tags: Union[bool, dict[str, bool]], column_names: list[str]):
             if isinstance(use_standard_tags, bool):
                 use_standard_tags = {col_name: use_standard_tags for col_name in column_names}
             else:
                 use_standard_tags = {**{col_name: True for col_name in column_names}, **use_standard_tags}
+            return use_standard_tags
 
-            self._schema = TableSchema(column_names=column_names,
-                                       logical_types=parsed_logical_types,
-                                       index=index,
-                                       time_index=time_index,
-                                       validate=validate,
-                                       use_standard_tags=use_standard_tags,
-                                       **kwargs)
+        if validate:
+            _validate_accessor_params(self._dataframe, index, time_index, logical_types, schema, use_standard_tags)
 
-            self._set_underlying_index()
-            if self._schema.time_index is not None:
-                self._sort_columns(already_sorted)
+        existing_logical_types = {}
+        existing_col_descriptions = {}
+        existing_col_metadata = {}
+        existing_use_standard_tags = {}
+        existing_semantic_tags = {}
+        existing_col_origins = {}
+        
+        if schema:  # pull schema parameters
+            name = name or schema.name
+            index = index or schema.index
+            time_index = time_index or schema.time_index
+            table_metadata = table_metadata or schema.metadata
```

Comment:
Should these objects that may be coming from another DataFrame's schema be deepcopied before being passed into a new DataFrame at ww init?

Basically, my fear is that using the same `table_metadata` object that's coming from a `schema` means that any changes to the resulting DataFrame's woodwork metadata would get propogated back to the DataFrame where this metadata originally came from

Reply:
I added a `test_init_with_partial_schema_metadata_deep_copy` to show that changes don't propagate